independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/multipart/EntityPartImpl.java

```diff
@@ -0,0 +1,72 @@
+package org.jboss.resteasy.reactive.server.multipart;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.Optional;
+
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.*;
+
+import org.jboss.resteasy.reactive.common.util.UnmodifiableMultivaluedMap;
+
+public class EntityPartImpl implements EntityPart {
+    private final String name;
+    private final MultivaluedMap<String, String> headers;
+    private final MediaType mediaType;
+    private final String fileName;
+    private final InputStream content;
+
+    public EntityPartImpl(FormValue value, String controlName) {
+        this.name = controlName;
+        this.headers = new UnmodifiableMultivaluedMap(value.getHeaders());
+        this.mediaType = MediaType.valueOf(value.getHeaders().getFirst("Content-Type"));
+        this.fileName = value.getFileName();
+
+        try {
+            if (value.isFileItem()) {
+                this.content = new FileInputStream(value.getFileItem().getFile().toFile());
+            } else {
+                this.content = new ByteArrayInputStream(value.getValue().getBytes(Charset.defaultCharset()));
+            }
+        } catch (IOException e) {
+            throw new MultipartPartReadingException(e);
+        }
+    }
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @Override
+    public Optional<String> getFileName() {
+        return Optional.ofNullable(this.fileName);
+    }
+
+    @Override
+    public InputStream getContent() {
+        return this.content;
+    }
+
+    @Override
+    public <T> T getContent(Class<T> type)
+            throws IllegalArgumentException, IllegalStateException, IOException, WebApplicationException {
```

Comment:
For this method and the next one, it appears we're only allowed to invoke it once (see javadoc). 

Probably the body should invoke `MultipartSupport.getConvertedFormAttribute` (same for the next method)

Reply:
With a special fast-case for `String`, `char[]`, `CharSequence` perhaps, since we already have that in memory.